cloud/deploy/bundle.go

```diff
@@ -0,0 +1,247 @@
+		err := os.Remove(tarFilePath)
+		if err != nil {
+			fmt.Println("\nFailed to delete tar file: ", err.Error())
+			fmt.Println("\nPlease delete the tar file manually from path: " + tarFilePath)
+		}
+	}()
+
+	// Generate the bundle tar
+	err := fileutil.Tar(bundlePath, tarFilePath, prependBaseDir)
+	if err != nil {
+		return "", err
+	}
+
+	tarFile, err := os.Open(tarFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer tarFile.Close()
+
+	versionID, err := azureUploader(uploadURL, tarFile)
+	if err != nil {
+		return "", err
+	}
+
+	return versionID, nil
+}
+
+func getDeployment(organizationID, deploymentID string, platformCoreClient astroplatformcore.CoreClient) (*astroplatformcore.Deployment, error) {
+	resp, err := platformCoreClient.GetDeploymentWithResponse(context.Background(), organizationID, deploymentID)
+	if err != nil {
+		return nil, err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.JSON200, nil
+}
+
+func createBundleDeploy(organizationID string, input *DeployBundleInput, deployGit *astrocore.DeployGit, coreClient astrocore.CoreClient) (*astrocore.Deploy, error) {
+	request := astrocore.CreateDeployRequest{
+		Description:     &input.Description,
+		Type:            astrocore.CreateDeployRequestTypeBUNDLE,
+		BundleMountPath: &input.MountPath,
+		BundleType:      &input.BundleType,
+	}
+	if deployGit != nil {
+		request.Git = &astrocore.CreateDeployGitRequest{
+			Provider:   astrocore.CreateDeployGitRequestProvider(deployGit.Provider),
+			Repo:       deployGit.Repo,
+			Account:    deployGit.Account,
+			Path:       deployGit.Path,
+			Branch:     deployGit.Branch,
+			CommitSha:  deployGit.CommitSha,
+			CommitUrl:  fmt.Sprintf("https://github.com/%s/%s/commit/%s", deployGit.Account, deployGit.Repo, deployGit.CommitSha),
+			AuthorName: deployGit.AuthorName,
+		}
+	}
+	resp, err := coreClient.CreateDeployWithResponse(context.Background(), organizationID, input.DeploymentID, request)
+	if err != nil {
+		return nil, err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return resp.JSON200, nil
+}
+
+func finalizeBundleDeploy(organizationID, deployID, tarballVersion string, input *DeployBundleInput, coreClient astrocore.CoreClient) (*astrocore.UpdateDeployResponse, error) {
+	request := astrocore.UpdateDeployRequest{
+		BundleTarballVersion: &tarballVersion,
+	}
+	resp, err := coreClient.UpdateDeployWithResponse(context.Background(), organizationID, input.DeploymentID, deployID, request)
+	if err != nil {
+		return nil, err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return resp, nil
+}
+
+func retrieveLocalGitMetadata(bundlePath string) *astrocore.DeployGit {
+	if git.HasUncommittedChanges() {
+		logrus.Warn("Local repository has uncommitted changes, skipping Git metadata retrieval")
+		return nil
+	}
```

Comment:
I don't know where this ultimately ends up, but instead of having no git metadata could we have the `-dirty` flag appended like `git describe does`:

```bash
$ git describe --tags --dirty
v1.27.0-12-gd79dc210-dirty
```

Reply:
The intention of the Git metadata is to correlate a deploy with a commit that users can look up to see what the contents are/were (like we have done for GitHub integration), so we don't want to stamp the deploy with metadata that only relates to a dirty state on a local machine.